packages/blade/src/components/Link/BaseLink/BaseLink.tsx

```diff
@@ -55,6 +55,7 @@ type BaseLinkAnchorVariantProps = BaseLinkPropsWithOrWithoutIcon & {
   target?: string;
   rel?: string;
   isDisabled?: undefined;
+  type?: never;
```

Comment:
I don't think we shall expose this prop. Since Link buttons ideally will only be used as href or `type=button` we can just pass it directly when we are rendering the component inside Link.

Reply:
A developer might want to use any of these values for type though
```ts
type?: 'button' | 'reset' | 'submit';
```

Have kept it as `never` for the anchor variant since type shouldn't be passed to anchor tag

Reply:
+1 to @kamleshchandnani. 

@PranjalMaithani where do you forsee a Link that is semantically a button to be used as `reset` or `submit`? Imo this exposes us to a risk of improper usage of the Link component

Reply:
Keeping it similar to Button props. Can omit `reset` and `submit` if they shouldn't be an option